File: slack/slack-notifier.py
#!/usr/bin/env python

import requests
import json
import os.path

if __name__=='__main__':

    path = ''
    webhook_url = ''

    with open('../env.json') as f:
        env_data = json.load(f)

        os_path = os.path.dirname(__file__);

        path = os_path + '/' + env_data['dataPathDev']
        webhook_url = env_data['webhookUrl']

    with open(path, 'r+') as f:
        menu_data = json.load(f)
        
        menu_text = ""
        for location in menu_data['locations']:
            menu_text += '*' + location['location'] + '*\n'
            menu_text += location['menu'] + "\n"

        menu_url = menu_data['url']

        attachments = [{
            "fallback": "Today's menu",
            "color": "good",
            "title": "Today's menu",
            "title_link": menu_url,
            "fields": [
                {
                    "value": menu_text,
                    "short": False
                }
            ],
            "mrkdwn_in": ["fields"]
        }]

        payload = { 'attachments': attachments }

        r = requests.post(webhook_url, data=json.dumps(payload))

File: slack/slack-notifier.js
const request = require('request');

require('dotenv').config();

// The webhook to post to
const webhookUrl = process.env.WEBHOOK_URL;

// Today’s menu data
const menuData = require(process.env.DATA_PATH + 'today.json');
const menuUrl = menuData.url;

// Build the menu data into a formatted string
const menuText =
  menuData.locations
    .map(({menu, location}, i) => `*${location}*\n${menu}`)
    .join('\n');

// Slack attachments
const attachments = [{
  fallback: 'Today’s menu',
  color: 'good',
  title: 'Today’s menu',
  title_link: menuUrl,
  fields: [
    {
      value: menuText,
      short: false
    }
  ],
  mrkdwn_in: [ 'fields' ]
}];

// Request options
const options = {
  method: 'post',
  body: { attachments },
  json: true,
  url: webhookUrl
};

// Send the data
request(options, (err, res) => {
  if (err) {
    console.log('Error sending data to Slack');
    console.log(err);
  }
});
